Firestore/core/src/bundle/bundle_document.h

```diff
@@ -18,21 +18,26 @@
 
 #include <utility>
 
+#include "Firestore/core/src/bundle/bundle_element.h"
 #include "Firestore/core/src/model/document.h"
 
 namespace firebase {
 namespace firestore {
 namespace bundle {
 
 /** Represents a document that was saved to a bundle. */
-class BundleDocument {
+class BundleDocument : public BundleElement {
  public:
   BundleDocument() = default;
 
   explicit BundleDocument(model::Document document)
       : document_(std::move(document)) {
   }
 
+  BundleElementType ElementType() const override {
+    return BundleElementType::Document;
```

Comment:
Optional: consider using `snake_case` for the name of this getter. It would indicate that this is a simple getter, not a more complicated function. I'm primarily mentioning this because this class already uses this convention with `key` below.